.github/pull_request_template.md

```diff
@@ -0,0 +1,19 @@
+<!--
+If you are a first-time contributor to OpenMC, please have a look at our
+contributing guidelines:
+https://github.com/openmc-dev/openmc/blob/develop/CONTRIBUTING.md
+-->
+
+# Description
+
+Please include a summary of the change and which issue is fixed if applicable. Please also include relevant motivation and context.
+
+Fixes # (issue)
+
+# Checklist
+
+- [ ] I have performed a self-review of my own code
+- [ ] I have run [clang-format](https://docs.openmc.org/en/latest/devguide/styleguide.html#automatic-formatting) on any C++ source files (if applicable)
+- [ ] I have followed the [style guidelines](https://docs.openmc.org/en/latest/devguide/styleguide.html#python) for Python source files (if applicable)
+- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
+- [ ] I have made corresponding changes to the documentation (if applicable)
```

Comment:
Should we suggest running regression/unit tests locally before submitting a PR? For a while, I relied on github to do my testing, and that's kind of a waste of compute resources.

Reply:
FWIW I like that idea, but I think we might need to add some helping documentation on how to get tests to pass locally since that can often be a pain.

Reply:
Perhaps we can suggest doing this (as a comment in the template) but I don't know that I'd want it to be part of the checklist that everyone is expected to do. That significantly raises the barrier for someone wanting to submit a simple PR.

Either way, we should improve the docs about getting tests to pass locally.

Reply:
Makes sense to me, yeah, that can be a little tricky at first.

Reply:
Just added a comment in the template to address this